src/components/Dropdown/common/DropdownTag.jsx

```diff
@@ -22,41 +29,182 @@ class DropdownTag extends Component {
   constructor(props) {
     super(props);
 
-    this.state = { tag: props.tag, isHighlighted: false };
+    this.state = {
+      tag: props.tag,
+      isHighlighted: false,
+      tagProperty: {},
+    };
     this.onRemove = this.onRemove.bind(this);
     this.onClick = this.onClick.bind(this);
+    this.handleSetTagProperty = this.handleSetTagProperty.bind(this);
+    this.toggleHighligth = this.toggleHighligth.bind(this);
+  }
+
+  componentWillMount() {
+    const { t, name, tag } = this.props;
+
+    if (name === 'filter') {
+      switch (tag.relevanceId) {
+        case RESOURCE_FILTER_CORE.id:
+          this.setState({
+            tagProperty: {
+              ...RESOURCE_FILTER_CORE,
+              name: t(RESOURCE_FILTER_CORE.name),
+            },
+          });
+          break;
+        case RESOURCE_FILTER_SUPPLEMENTARY.id:
+          this.setState({
+            tagProperty: {
+              ...RESOURCE_FILTER_SUPPLEMENTARY,
+              name: t(RESOURCE_FILTER_SUPPLEMENTARY.name),
+            },
+          });
+          break;
+        default:
+          break;
+      }
+    }
+    if (name === 'topics') {
+      this.setState({
+        tagProperty: tag.primary
+          ? {
+              ...RESOURCE_TOPICS_PRIMARY,
+              name: t(RESOURCE_TOPICS_PRIMARY.name),
+            }
+          : {},
+      });
+    }
   }
 
   componentWillReceiveProps(nextProps) {
     if (nextProps.tag !== this.props.tag) {
       this.setState({ tag: nextProps.tag });
     }
+    if (nextProps.tag.primary !== this.props.tag.primary) {
+      if (nextProps.tag.primary) {
+        this.setState({
+          tagProperty: {
+            ...RESOURCE_TOPICS_PRIMARY,
+            name: this.props.t(RESOURCE_TOPICS_PRIMARY.name),
+          },
+          isHighlighted: false,
+        });
+      }
+    }
   }
 
   onClick() {
-    this.setState(({ isHighlighted }) => ({ isHighlighted: !isHighlighted }));
+    const { tag } = this.state;
+    const { name } = this.props;
+
+    if (name === 'topics' && !tag.primary) {
+      this.toggleHighligth();
+    }
   }
 
   onRemove(e) {
-    const { onRemoveItem, tag } = this.props;
+    const { onRemoveItem, name, tag } = this.props;
     e.preventDefault();
     e.stopPropagation();
     if (onRemoveItem) {
-      onRemoveItem({ ...tag });
+      onRemoveItem(tag, name);
     }
   }
 
+  handleSetTagProperty(e) {
```

Comment:
Bittelitt bedre å bruke find her? Ala noe sånt: 

`const item = tagProperties.find(it => it.id === value);
if(item) {
  this.setState({tagProperty: item});
  handlePopupClick({...this.state.tag, relevanceId: item.id})
}`